lager.go

```diff
@@ -44,7 +44,7 @@ func (f Lager) Logger(component string) (lager.Logger, *lager.ReconfigurableSink
 	if f.writerSink == nil {
 		f.writerSink = os.Stdout
 	}
-	sink := lager.NewReconfigurableSink(lager.NewWriterSink(f.writerSink, lager.DEBUG), minLagerLogLevel)
+	sink := lager.NewReconfigurableSink(lager.NewPrettySink(f.writerSink, lager.DEBUG), minLagerLogLevel)
```

Comment:
Probably better to make this configurable.

Maybe just do the same thing they did here, for consistency?:

https://github.com/cloudfoundry/guardian/blob/6d76846840e7b00431e5418f5cf176a6aec5904a/guardiancmd/lager_flag.go#L22

For now I would keep the default of `unix-epoch` here - if we want to make human-readable timestamps the default we can do that somewhere in `bin/` and propagate the setting to every component: https://github.com/concourse/concourse/tree/master/bin

Reply:
I thought about to add an option actually. But I think it makes sense to use human readable timestamp in log by default. Is there any special consideration to print `unit-epoch` by default? Coz I never see a system print logs with `unix-epoch` timestamp. I would guess 99% of users would like to read YYYY-MM-DD formatted timestamp.

Reply:
My main concern is with backwards-compatibility, but I guess the next version is 5.0 so maybe we can change the default anyway and just put it in the release notes.

We'll have to modify `bin`, though, to change Garden's default. I can do that after merging this in.

Reply:
There are a few places in `concourse/concourse` use `WriterSink`. I planed to open a separate PR for that as well, if you can do that, that would be awesome. :)

I agree this can be merged to only 5.0. So, will you still want me to update this change?

By the way, we are trying to using Concourse as a big CI platform, so we'll open more PRs in future. :)

Reply:
@vito I tested this patch in my local using `docker-compose up`, I did `go mod replace` to inject my local `flag`. This this patch, `atc`, `tsa` and `worker` are now printing YMD timestamp:

    worker_1  | {"timestamp":"2019-01-29T09:11:38.053227500Z","level":"debug","source":"worker","message":"worker.beacon.keepalive.tick","data":{"session":"7.3"}}
    web_1     | {"timestamp":"2019-01-29T09:11:40.338770100Z","level":"info","source":"tsa","message":"tsa.connection.channel.command.heartbeat.start","data":{"command":"forward-worker","remote":"172.18.0.4:57362","session":"1.3.1.84","worker-address":"127.0.0.1:36843","worker-platform":"linux","worker-tags":""}}
    
But `guardian` still shows `unix-epoch` timestamp:

    worker_1  | {"timestamp":"1548753100.342236519","source":"guardian","message":"guardian.list-containers.starting","log_level":1,"data":{"session":"198"}}
    worker_1  | {"timestamp":"1548753100.342351198","source":"guardian","message":"guardian.list-containers.finished","log_level":1,"data":{"session":"198"}}

I replaced all `WriterSink` with `PrettySink` in `concourse/concourse` repo which doesn't help at all.

Then I found a possible place to modify is in `concourse/guardian` repo, but I don't see `guardian` module in `concourse/concourse/go.mod`. Do you have idea where to update `guardian`'s logger sink?

Reply:
I believe this'll need to be set as a flag here:

https://github.com/concourse/concourse/blob/master/bin/cmd/concourse/worker_linux.go#L69

We run Guardian via its `gdn` binary; it's not compiled in to Concourse.

You could probably just pass something like `"--time-format", cmd.Logger.TimeFormat,`.

Reply:
@vito I just tried and it worked. I added a line there:

    "--time-format", "rfc3339",

and now `guardian` log has consistent time format:

    worker_1  | {"timestamp":"2019-01-30T00:07:26.428451000Z","level":"info","source":"guardian","message":"guardian.list-containers.starting","data":{"session":"45"}}
    worker_1  | {"timestamp":"2019-01-30T00:07:26.428609100Z","level":"info","source":"guardian","message":"guardian.list-containers.finished","data":{"session":"45"}}

I'll create a PR for `concourse/concourse` today.

Reply:
Hey,

I noticed that this also changes the `log_level` to instead of displaying a number, displaying a text:

```
{"timestamp":"1549306000.436143160","source":"my-app","message":"my-app.ahuhs","log_level":1,"data":{}}
```

vs

```
{"timestamp":"2019-02-04T18:46:55.949682000Z","level":"info","source":"my-app","message":"my-app.ahuhs","data":{}}
```

(see https://github.com/cloudfoundry/lager/blob/master/models.go)

This seems to be a bit different from what is done in guardian (https://github.com/cloudfoundry/guardian/blob/6d76846840e7b00431e5418f5cf176a6aec5904a/guardiancmd/lager_flag.go#L22), where it seems to only change the time format. Is this intended?

thx!

Reply:
Oh, actually, I'm wrong, `guardian` indeed changes the whole writer to `pretty` as well: https://github.com/cloudfoundry/guardian/blob/6d76846840e7b00431e5418f5cf176a6aec5904a/guardiancmd/lager_flag.go#L42

Sorry for the confusion!